### PR TITLE
[WIP]Add metrics and postprocess

### PR DESCRIPTION
**Pull request by zhangyubo0722**

### CODE REVIEW

File: deploy/python/postprocess.py [hunk 1]
```diff
@@ -346,6 +346,58 @@ def __call__(self, batch_preds, file_names=None):
...
+            ["RosyCheeks", "红润的双颊"], ["Sideburns", "连鬓胡子"], ["Smiling", "微笑"],
+            ["StraightHair", "直发"], ["WavyHair", "卷发"],
+            ["WearingEarrings", "戴着耳环"], ["WearingHat", "戴着帽子"],
+            ["WearingLipstick", "涂了唇膏"], ["WearingNecklace", "戴着项链"],
+            ["WearingNecktie", "戴着领带"], ["Young", "年轻人"]
+        ]
+        gender_list = [["Male", "男性"], ["Female", "女性"]]
+        age_list = [["Young", "年轻人"], ["Old", "老年人"]]
+        batch_res = []
+        if self.convert_cn:
```
Comment (cuicheng01) [inline deploy/python/postprocess.py]: 一行代码

File: ppcls/data/postprocess/attr_rec.py [hunk 2]
```diff
@@ -171,6 +173,66 @@ def __call__(self, x, file_names=None):
...
+            ["RosyCheeks", "红润的双颊"], ["Sideburns", "连鬓胡子"], ["Smiling", "微笑"],
+            ["StraightHair", "直发"], ["WavyHair", "卷发"],
+            ["WearingEarrings", "戴着耳环"], ["WearingHat", "戴着帽子"],
+            ["WearingLipstick", "涂了唇膏"], ["WearingNecklace", "戴着项链"],
+            ["WearingNecktie", "戴着领带"], ["Young", "年轻人"]
+        ]
+        gender_list = [["Male", "男性"], ["Female", "女性"]]
+        age_list = [["Young", "年轻人"], ["Old", "老年人"]]
+        batch_res = []
+        if self.convert_cn:
```
Comment (cuicheng01) [inline ppcls/data/postprocess/attr_rec.py]: 同上

File: ppcls/metric/metrics.py [hunk 3]
```diff
@@ -218,6 +218,55 @@ def avg_info(self):
         return result
 
 
+class MultilabelMeanAccuracy(nn.Layer):
+    def __init__(self, class_num=40):
```
Comment (cuicheng01) [inline ppcls/metric/metrics.py]: 这里的class_num最好不要放到参数里，可以在forward中根据x的大小得到

File: ppcls/metric/metrics.py [hunk 4]
```diff
@@ -218,6 +218,55 @@ def avg_info(self):
...
+
+    def forward(self, x, label):
+        if isinstance(x, dict):
+            x = x["logits"]
+        x = F.sigmoid(x)
+        label = label[:, 0, :]
+        for i in range(len(x)):
+            self.gt_all_score_list.append(x[i].numpy())
+            self.gt_label_score_list.append(label[i].numpy())
+        return {}
```
Comment (cuicheng01) [inline ppcls/metric/metrics.py:242]: 这里return {}是啥意思

File: ppcls/metric/metrics.py [hunk 5]
```diff
@@ -218,6 +218,55 @@ def avg_info(self):
...
+    def avg(self):
+        return self.max_acc
+
+    @property
+    def avg_info(self):
+        max_acc = 0.
+        result = ""
+        gt_all_score_list = np.array(self.gt_all_score_list)
+        gt_label_score_list = np.array(self.gt_label_score_list)
+        for i in range(10):
```
Comment (cuicheng01) [inline ppcls/metric/metrics.py]: 这个10，以及下边的0.05、0.4是需要写到__init__中的

**Review by cuicheng01 — APPROVED** [review]
